native_client/deepspeech.cc

```diff
@@ -291,6 +290,19 @@ DS_CreateModel(const char* aModelPath,
   return DS_ERR_OK;
 }
 
+unsigned int
+DS_GetModelBeamWidth(ModelState* aCtx)
+{
+  return aCtx->beam_width_;
+}
+
+int
+DS_SetModelBeamWidth(ModelState* aCtx, unsigned int aBeamWidth)
```

Comment:
So we overwrite the default value that was read from the model. I guess that's fine, user could still `DS_GetModelBeamWidth` right after loading the model to get the default value, but we may want to make it clear in the documentation ?

Reply:
Have you looked at the `DS_GetModelBeamWidth` documentation? Do you think I need to add something else to it?

Reply:
Well, the doc says it returns the value used by the model, and that unless you call set it will be the default, which is true. I just fear that people will not pay attention to it since the Set call is destructive.

Maybe a big `**WARNING** **Overwrite** the default model's beam with value with your own value. **Use with caution**.` would be more clear in the mind of people, in the documentation of `DS_SetModelBeamWidth`.

Reply:
I don't see the point though. It's not like the set call is destructive on the filesystem, it's just for that Model instance. If you try it out and it's not working well, change the code and run again. I guess I don't understand why you think this needs such a big warning.

Reply:
I guess I'm biaised by replying on Discourse :). Let's not waste more time on that, we still have time to improve if needed.